go/nbs/table_set.go

```diff
@@ -21,9 +21,18 @@ func newTableSet(persister tablePersister) tableSet {
 
 // tableSet is an immutable set of persistable chunkSources.
 type tableSet struct {
-	novel, compacted, compactees, upstream chunkSources
-	p                                      tablePersister
-	rl                                     chan struct{}
+	// novel chunkSources contain chunks that have not yet been pushed upstream
+	novel chunkSources
+
+	// When Compact() conjoins upstream tables into larger tables, the results are stored in |compacted| until the next call to Flatten(). The tables that were conjoined are put into |compactees|, so they can be correctly handled during a Rebase()
```

Comment:
I was kind of more looking for a description of the relationship between compacted, compactees and upstream. kind of explaining that if compacted is non-empty, then compactees will have a corresponded set of tables which comprised tables in compacted and upstream will "lack" the tables in compactees (if one actually "looked" upstream).